sherpa/conftest.py

```diff
@@ -556,50 +572,42 @@ def old_numpy_printing():
         np.set_printoptions(legacy=oldopts['legacy'])
 
 
-PLOT_BACKENDS = [dummy_backend]
-if pylab_backend is not None:
-    PLOT_BACKENDS.append(pylab_backend)
-
-
-@pytest.fixture(params=PLOT_BACKENDS)
-def override_plot_backend(request):
+@pytest.fixture(params=PLOT_BACKENDS.keys())
+def all_plot_backends(request):
     """Override the plot backend for this test
 
     Runs the test with the given plot backend and then restores the
     original value.
 
-    Note that this does not reload the ui layers, which will have
-    retain reference to the original backend throughout.
-
+    Note that this does not reload the ui layers, which will
+    retain a reference to the original backend throughout.
     """
+    with TemporaryPlottingBackend(request.param):
+        yield
+        if request.param == 'pylab':
+            from matplotlib import pyplot as plt
+            plt.close(fig="all")
 
-    old = sherpa.plot.backend
-
-    changed = old.name != request.param.name
-    if changed:
-        sherpa.plot.backend = request.param
-        sherpa.astro.plot.backend = request.param
 
-    yield
-    if changed:
-        sherpa.plot.backend = old
-        sherpa.astro.plot.backend = old
+@pytest.fixture(params=FUNCTIONAL_PLOT_BACKENDS)
+def plot_backends(request):
+    """Override the plot backend for this test (only for functional backends)
 
+    This fixtures runs tests for all backens that actually produce plots, i.e.
+    not for any type of dummy.
 
-@pytest.fixture(autouse=True, scope="session")
-def cleanup_pylab_backend():
-    """Ensure that the pylab backend has closed down all windows.
+    Runs the test with the given plot backend and then restores the
+    original value.
 
-    This is related to https://github.com/The-Compiler/pytest-xvfb/issues/11
-    and the idea is to ensure that all matplotlib windows are closed at the
-    end of the tests.
+    Note that this does not reload the ui layers, which will
+    retain a reference to the original backend throughout.
     """
 
     yield
```

Comment:
I think something must have gone wrong with a rebase here since this test - `plot_backends` - does not include the `with TemoraryPlttnigBackend(request.param)` code that the test description says should be here.

I'm guessing it was doe to a rebase to include #1516 - it's a problem I found when I tried rebasing this PR but I think I got it finally done in #1550 (although that series now fails the tests unlike this code, so I obviously missed some other issue).